static/app/views/dashboards/dashboardLegendUtils.tsx

```diff
@@ -0,0 +1,204 @@
+        router.replace({
+          query: {
+            ...location.query,
+            unselectedSeries: newLegendQuery,
+          },
+        });
+      }
+    } else if (Array.isArray(location.query.unselectedSeries)) {
+      let isInQuery = false;
+      newLegendQuery = location.query.unselectedSeries.map(widgetLegend => {
+        if (widgetLegend.includes(widget.id!)) {
+          isInQuery = true;
+          // names of legend options are stored as seriesName:widgetId and are stored in
+          // query param as widgetId-seriesName-seriesName2-...
+          return this.encodeLegendQueryParam(widget, selected);
+        }
+        return widgetLegend;
+      });
+
+      !isInQuery
+        ? router.replace({
+            query: {
+              ...location.query,
+              unselectedSeries: [
+                ...location.query.unselectedSeries,
+                this.encodeLegendQueryParam(widget, selected),
+              ],
+            },
+          })
+        : router.replace({
+            query: {
+              ...location.query,
+              unselectedSeries: newLegendQuery,
+            },
+          });
+    } else {
+      router.replace({
+        query: {
+          ...location.query,
+          unselectedSeries: [
+            location.query.unselectedSeries,
+            this.encodeLegendQueryParam(widget, selected),
+          ],
+        },
+      });
+    }
+  }
+
+  // sets unselected legend options by the legend query param
+  getLegendUnselected(widget: Widget) {
+    const location = this.location;
+
+    return location.query.unselectedSeries
+      ? this.decodeLegendQueryParam(widget)
+      : this.widgetRequiresLegendUnselection(widget) &&
+          this.organization.features.includes('dashboards-releases-on-charts')
+        ? {[this.encodeSeriesNameForLegend('Releases', widget.id)]: false}
+        : {};
+  }
+
+  widgetRequiresLegendUnselection(widget: Widget) {
+    return (
+      widget.displayType === DisplayType.AREA || widget.displayType === DisplayType.LINE
+    );
+  }
+
+  formatLegendDefaultQuery(widgetId?: string) {
+    return this.organization.features.includes('dashboards-releases-on-charts')
+      ? `${widgetId}-Releases`
+      : `${widgetId}-`;
+  }
+
+  // going from selected to query param
+  encodeLegendQueryParam(widget: Widget, selected: Record<string, boolean>) {
+    return (
+      widget.id +
+      '-' +
+      Object.keys(selected)
+        .filter(key => !selected[key])
+        .map(series => this.decodeSeriesNameForLegend(series))
+        .join('-')
+    );
+  }
+
+  // going from query param to selected
+  decodeLegendQueryParam(widget: Widget) {
+    const location = this.location;
+    return decodeList(location.query.unselectedSeries).reduce((acc, legend) => {
+      const legendValues = legend.split('-');
+      const widgetId = legendValues[0];
+      const seriesNames = legendValues.splice(1);
```

Comment:
```suggestion
      const [widgetId, ...seriesNames] = legend.split('-');
```

If you stick to using dash. Otherwise

```jsx
const [widgetId, seriesNameString] = legend.split('-');
const seriesNames = seriesNameString.split(',');
```

If you go with commas